### ISSUE TITLE
resolved

### ISSUE THREAD

**Issue by GISScience** [FH-Potsdam/shifted-maps#54]
i have created a plugin

**Comment by lennerd** [comment]
I'm not sure I can follow. Did you create a fork of Shifted Maps or do you try to use it as a library? I also have absolutely no experience with `elastic/kibana`, but the script paths inside the package.json pointing to a directory outside the root folder look strange to me, TBH.

Fyi: Shifted Maps is primary a standalone website built with [`next`](https://nextjs.org/). For example this means that there is no built in way to built Shifted Maps source files for using it in another project right out of the box. The current built process is solely for creating a Node based web server to be deployed somewhere. Typescript is also used via a plugin called `next-typescript`. I never compiled the source code with the `typescript` CLI directly, so I'm cannot guarantee it will work out of the box either.

Either way looking at the error message you posted it looks like you're trying to have different typings with the same type definition (s. `Duplicate identifier`). There are also some problems with missing modules (s. `not found`).

Both errors seem to have nothing to do with Shifted Maps source code, so I would like to close this issue for now. Feel free to post more information if you feel this errors correspond to the Shifted Maps source code (pages, hooks, utils and components) directly.

**Comment by GISScience** [comment]
error is with respect to typescript... i am unable to compile ts and tsx files using typescript modules.

any alternative to compile it offline ,(without using babel or tsc) ??

i have removed next loader and added babel plugin helper loader.

**Comment by lennerd** [comment]
What do you mean with "compile it offline"? Typescript always needs to be compiled to pure javascript before beeing used inside the browser. There is no way around it.

You can try to remove alle `next` related code and use the components from Shifted Maps directly, for example the map or visualization components. Do you have any examples from your code using React components from this library? Or maybe more context and information on how you plan to use Shifted Maps or parts of it in your own application?

**Comment by GISScience** [comment]
thanks for your quick response and help...
kibana is leading dashboard creation application which is used by thousand of users..
my goal is to use shifted-map "map and visualization components" in custom kibana plugin which i have created.

What do you mean with "compile it offline"? :-  i mean using some thired party tool like tsc compiler , can i transcribe / compile shifted map ts and tsc files.

![Untitled](https://user-images.githubusercontent.com/43044931/67635220-83df5c80-f8ea-11e9-9f1b-a6df9272ec3e.png)

in above image , one error with respect to one-module of 'react-leaflet'.

in total i am getting so mane errors

![image](https://user-images.githubusercontent.com/43044931/67635246-be48f980-f8ea-11e9-997f-c3313be73f4e.png)


but if i remove typescript from plugin by "yarn remove typescript" , all errors got resolved... but ts and tsx compilation didnot happen(hence no use).

I have also removed next-server based rendering and calling map.ts from react component

code is below:- 

im```
port { NextStatelessComponent } from 'next';
import dynamic from 'next/dynamic';
import { useCallback } from 'react';
import React from 'react';
import { IMapView } from '../Visualisation/Visualisation';
import { DiaryData } from '../../stores/Diary';
import { VIEW } from '../../stores/UIStore';

const DynamicVisualisation = dynamic({
  loader: () => import('../Visualisation/Visualisation'),
  loading: () => null,
  ssr: false,
});

interface IProps {
  data: DiaryData;
  view?: VIEW;
  timeSpan?: ReadonlyArray<number>;
  mapView?: IMapView;
}

const Map: NextStatelessComponent<IProps> = props => {
  const { data, view, timeSpan, mapView } = props;

  const handleViewChange = useCallback((view?: VIEW) => {
    const query: { view?: string } = {
     
    };

    if (view != null) {
      query.view = VIEW[view].toLowerCase();
    } else {
      delete query.view;
    }

  }, []);


  return (
    <DynamicVisualisation
      data={data}
    />
  );
};

Map.getInitialProps = ({ query }) => {
  let timeSpan: ReadonlyArray<number> | undefined;
  let view: VIEW | undefined;
  let mapView: IMapView | undefined;

  if (typeof query.timeSpan === 'string') {
    const [start, end] = query.timeSpan.split('-');

    if (start != null && end != null) {
      timeSpan = [Number(start), Number(end)];
    }
  }

  if (typeof query.view === 'string') {
    view = VIEW[query.view.toUpperCase()];
  }

  if (typeof query.center === 'string' && typeof query.zoom === 'string') {
    const [lat, lng] = query.center.split(',');

    if (lat != null && lng != null) {
      mapView = {
        center: [Number(lat), Number(lng)],
        zoom: Number(query.zoom),
      };
    }
  }
```

**Comment by lennerd** [comment]
It‘s still very hard to follow.

Can you edit your post and wrap your code examples in three backticks (code blocks in Markdown)? I already edited your first post to be more readable. So you can find an example there.

It would also help a lot if you could provide a repository with your code and all the necessary steps to reproduce your errors.
